### ISSUE TITLE
tf.nn.elu returns zero gradient for negative float64 input where analytic derivative is finite

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#124833]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.21.0

### Custom code

Yes

### OS platform and distribution

 Linux Ubuntu 22.04

### Python version

Python 3.13

### Current behavior?

`tf.nn.elu` returns an incorrect first derivative for a negative float64 input.

For ELU with default `alpha=1`, the negative branch is `exp(x) - 1`, so the derivative for `x < 0` should be `exp(x)`. At `x = -40.0`, the analytic high-precision reference is approximately `4.248354255291589e-18`, which is a finite normal float64 value.

However, TensorFlow autodiff returns `0.0` for the gradient at this input. PyTorch computes the expected finite value for the same derivative, so this appears to be a backward-pass issue rather than a float64 limitation.

Expected behavior?

The gradient of `tf.nn.elu` at `x = -40.0` should be close to `exp(-40.0)`, approximately `4.248354255291589e-18`, instead of `0.0`.

### Standalone code to reproduce the issue

```shell
import os
os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

import math
import tensorflow as tf

x = -40.0
truth = math.exp(x)
rtol = 1e-6

v = tf.Variable(x, dtype=tf.float64)

with tf.GradientTape() as tape:
    y = tf.nn.elu(v)

g = tape.gradient(y, v)

actual = float(g)
rel_err = abs(actual - truth) / abs(truth)

print(f"x: {x!r}")
print(f"tensorflow gradient: {actual!r}")
print(f"expected gradient: {truth!r}")
print(f"relative error: {rel_err:.3e}")

if rel_err > rtol:
    print("BUG REPRODUCED: tf.nn.elu gradient is wrong for negative float64 input")
else:
    print("not reproduced")
```

### Relevant log output

```shell
x: -40.0
tensorflow gradient: 0.0
expected gradient: 4.248354255291589e-18
relative error: 1.000e+00
BUG REPRODUCED: tf.nn.elu gradient is wrong for negative float64 input
```

**Comment by GodlyDonuts** [comment]
I reproduced this on TensorFlow 2.20.0 and am working on a focused fix with regression coverage. The issue is that the registered ELU gradient derives `exp(x)` from the rounded activation output (`output + 1`), which becomes zero once the output rounds to exactly `-1.0`.

**Comment by Venkat6871** [comment]
Hi @ALinrunrun ,
Thank you for raising your concern here.
A fix for this issue has been proposed in this [PR](https://github.com/tensorflow/tensorflow/pull/124902). Once it is merged, this issue will be resolved.

Hi @GodlyDonuts ,
Thank you for raising the PR to address this issue. We truly appreciate your initiative and the effort in contributing the fix.
